src/platform/packages/private/kbn-grid-layout/grid/grid_panel/drag_handle/index.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useCallback, useEffect, useImperativeHandle, useRef, useState } from 'react';
+
+import { GridLayoutStateManager } from '../../types';
+import { useGridLayoutEvents } from '../../use_grid_layout_events';
+import { DefaultDragHandle } from './default_drag_handle';
+
+export interface DragHandleApi {
+  setDragHandles: (refs: Array<HTMLElement | null>) => void;
+}
+
+export const DragHandle = React.forwardRef<
+  DragHandleApi,
+  {
+    gridLayoutStateManager: GridLayoutStateManager;
+    panelId: string;
+    rowIndex: number;
+  }
+>(({ gridLayoutStateManager, panelId, rowIndex }, ref) => {
+  const startInteraction = useGridLayoutEvents({
```

Comment:
The event handler here is only responsible for starting the interaction.
Additional listeners (like those for drag or resize actions) are attached dynamically during the `mouse down` event inside the `startInteraction` method.